packages/http-client-csharp/eng/scripts/Generate.ps1

```diff
@@ -2,16 +2,166 @@
+    Join-Path 'http' 'client' 'structure' 'multi-client'
+    Join-Path 'http' 'client' 'structure' 'renamed-operation'
+    Join-Path 'http' 'client' 'structure' 'two-operation-group'
+    Join-Path 'http' 'encode' 'bytes'
+    Join-Path 'http' 'encode' 'datetime'
+    Join-Path 'http' 'encode' 'duration'
+    Join-Path 'http' 'parameters' 'basic'
+    Join-Path 'http' 'parameters' 'body-optionality'
+    Join-Path 'http' 'parameters' 'collection-format'
+    Join-Path 'http' 'parameters' 'spread'
+    Join-Path 'http' 'payload' 'content-negotiation'
+    Join-Path 'http' 'payload' 'json-merge-patch'
+    Join-Path 'http' 'payload' 'media-type'
+    Join-Path 'http' 'payload' 'multipart'
+    Join-Path 'http' 'payload' 'pageable'
+    Join-Path 'http' 'resiliency' 'srv-driven'
+    Join-Path 'http' 'serialization' 'encoded-name' 'json'
+    Join-Path 'http' 'server' 'endpoint' 'not-defined'
+    Join-Path 'http' 'server' 'path' 'multiple'
+    Join-Path 'http' 'server' 'path' 'single'
+    Join-Path 'http' 'server' 'versions' 'not-versioned'
+    Join-Path 'http' 'server' 'versions' 'versioned'
+    Join-Path 'http' 'special-headers' 'conditional-request'
+    Join-Path 'http' 'special-headers' 'repeatability'
+    Join-Path 'http' 'type' 'array'
+    Join-Path 'http' 'type' 'dictionary'
+    Join-Path 'http' 'type' 'scalar'
+    Join-Path 'http' 'type' 'union'
+    Join-Path 'http' 'type' 'enum' 'extensible'
+    Join-Path 'http' 'type' 'enum' 'fixed'
+    Join-Path 'http' 'type' 'model' 'empty'
+    Join-Path 'http' 'type' 'model' 'flatten'
+    Join-Path 'http' 'type' 'model' 'usage'
+    Join-Path 'http' 'type' 'model' 'visibility'
+    Join-Path 'http' 'type' 'model' 'inheritance' 'enum-discriminator'
+    Join-Path 'http' 'type' 'model' 'inheritance' 'nested-discriminator'
+    Join-Path 'http' 'type' 'model' 'inheritance' 'not-discriminated'
+    Join-Path 'http' 'type' 'model' 'inheritance' 'recursive'
+    Join-Path 'http' 'type' 'model' 'inheritance' 'single-discriminator'
+    Join-Path 'http' 'type' 'property' 'additional-properties'
+    Join-Path 'http' 'type' 'property' 'nullable'
+    Join-Path 'http' 'type' 'property' 'optionality'
+    Join-Path 'http' 'type' 'property' 'value-types'
+)
+
+$cadlRanchLaunchProjects = @{}
+
+# Loop through all directories and subdirectories of the cadl ranch specs
+$directories = Get-ChildItem -Path "$specsDirectory/http" -Directory -Recurse
+foreach ($directory in $directories) {
+    if (-not (IsSpecDir $directory.FullName)) {
+        continue
+    }
+
+    $specFile = Join-Path $directory.FullName "main.tsp"
+    $subPath = $directory.FullName.Substring($specsDirectory.Length + 1)
+    $folders = $subPath.Split([System.IO.Path]::DirectorySeparatorChar)
+
+    if ($folders.Contains("azure")) {
+        continue
+    }
+
+    if ($folders.Contains("versioning")) {
+        continue # TODO: adopt versioning cadl ranch specs https://github.com/microsoft/typespec/issues/3965
+    }
```

Comment:
why not we just add those versioning specs to the above failing specs list?

Reply:
It is unclear what we plan to do with versioning information.  The other spec are ready to be consumed right away as we have a design / plan for them.